Campus-iOS/GradesComponent/Screen/GradesScreen.swift

```diff
@@ -26,63 +26,73 @@ struct GradesScreen: View {
-                FailedView(
-                    errorDescription: error.localizedDescription,
-                    retryClosure: vm.reloadGradesAndAverageGrades
-                )
             }
-        }
-        .task {
-            await vm.reloadGradesAndAverageGrades()
-        }
-        // Refresh whenever user authentication status changes
-        .onChange(of: self.refresh) { _ in
-            Task {
+            .task {
                 await vm.reloadGradesAndAverageGrades()
             }
-        }
-        // As LoginView is just a sheet displayed in front of the GradeScreen
-        // Listen to changes on the token, then fetch the grades
-        .onChange(of: self.vm.model.token ?? "") { _ in
-            Task {
-                await vm.reloadGradesAndAverageGrades()
+            // Refresh whenever user authentication status changes
+            .onChange(of: self.refresh) { _ in
+                Task {
+                    await vm.reloadGradesAndAverageGrades()
+                }
             }
-        }
-        .alert(
-            "Error while fetching Grades",
-            isPresented: $vm.hasError,
-            presenting: vm.state) { detail in
-                Button("Retry") {
-                    Task {
-                        await vm.reloadGradesAndAverageGrades()
-                    }
+            // As LoginView is just a sheet displayed in front of the GradeScreen
+            // Listen to changes on the token, then fetch the grades
+            .onChange(of: self.vm.model.token ?? "") { _ in
+                Task {
+                    await vm.reloadGradesAndAverageGrades()
                 }
-                
-                Button("Cancel", role: .cancel) { }
-            } message: { detail in
-                if case let .failed(error) = detail {
-                    if let apiError = error as? TUMOnlineAPIError {
-                        Text(apiError.errorDescription ?? "TUMOnlineAPI Error")
-                    } else {
-                        Text(error.localizedDescription)
+            }
+            .alert(
+                "Error while fetching Grades",
+                isPresented: $vm.hasError,
+                presenting: vm.state) { detail in
+                    Button("Retry") {
+                        Task {
+                            await vm.reloadGradesAndAverageGrades()
+                        }
+                    }
+                    
+                    Button("Cancel", role: .cancel) { }
+                } message: { detail in
+                    if case let .failed(error) = detail {
+                        if let apiError = error as? TUMOnlineAPIError {
+                            Text(apiError.errorDescription ?? "TUMOnlineAPI Error")
+                        } else {
+                            Text(error.localizedDescription)
+                        }
                     }
                 }
-            }
+        } else {
```

Comment:
please remove that if else clause - applying this before data is fetched, makes no sense